lib/auth/grpcserver.go

```diff
@@ -4252,14 +4253,14 @@ func (g *GRPCServer) SetInstaller(ctx context.Context, req *types.InstallerV1) (
 }
 
 // GetInstaller retrieves the installer script resource
-func (g *GRPCServer) GetInstaller(ctx context.Context, _ *empty.Empty) (*types.InstallerV1, error) {
+func (g *GRPCServer) GetInstaller(ctx context.Context, req *types.ResourceRequest) (*types.InstallerV1, error) {
 	auth, err := g.authenticate(ctx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	res, err := auth.GetInstaller(ctx)
+	res, err := auth.GetInstaller(ctx, req.Name)
 	if err != nil {
-		if trace.IsNotFound(err) {
+		if trace.IsNotFound(err) && req.Name == defaults.InstallerScriptName {
```

Comment:
This logic will basically allow the users to override the default installer resource if they want to? I think it's a good idea :+1: